waypoint_navigation/src/waypoint_nav.cpp

```diff
@@ -0,0 +1,317 @@
+#include "waypoint_navigation/waypoint_nav.h"
+
+WaypointsNavigation::WaypointsNavigation() : Node("waypoint_nav"), nav_time_(0), wp_num_(0), has_activate_(false)
+{
+  // Parameters
+  this->declare_parameter<std::string>("waypoints_file", "");
+  this->declare_parameter<std::string>("world_frame", "map");
+  this->declare_parameter<std::string>("robot_frame", "base_footprint");
+  this->declare_parameter<float>("min_dist_err", 0.3);
+  this->declare_parameter<float>("min_yaw_err", 0.3);
+
+  this->get_parameter("world_frame", world_frame_);
+  this->get_parameter("robot_frame", robot_frame_);
+  this->get_parameter("min_dist_err", min_dist_err_);
+  this->get_parameter("min_yaw_err", min_yaw_err_);
+
+  // Load YAML file
+  std::string waypoints_file = "";
+  this->get_parameter("waypoints_file", waypoints_file);
+  if (!readFile(waypoints_file))
+  {
+    throw std::runtime_error("Failed to read waypoints yaml file");
+  }
+  current_wp_ = pose_array_.poses.begin();
+  finish_pose_ = pose_array_.poses.end() - 1;
+  setWpOrientation();
+
+  // Publisher
+  wp_vis_pub_ = this->create_publisher<geometry_msgs::msg::PoseArray>("~/waypoints", 10);
+  wp_num_pub_ = this->create_publisher<std_msgs::msg::UInt16>("waypoint_num", 10);
+
+  // Service
+  start_server_ = create_service<std_srvs::srv::Trigger>(
+      "start_wp_nav", std::bind(&WaypointsNavigation::startNavCallback, this, _1, _2));
+  resume_server_ = create_service<std_srvs::srv::Trigger>(
+      "resume_nav", std::bind(&WaypointsNavigation::resumeNavCallback, this, _1, _2));
+
+  // Nav2 action
+  client_ptr_ = rclcpp_action::create_client<NavToPose>(this, "navigate_to_pose");
+  send_goal_opts_.goal_response_callback = std::bind(&WaypointsNavigation::responseCallback, this, _1);
+  send_goal_opts_.feedback_callback = std::bind(&WaypointsNavigation::feedbackCallback, this, _1, _2);
+  send_goal_opts_.result_callback = std::bind(&WaypointsNavigation::resultCallback, this, _1);
+  nav_status_ = rclcpp_action::ResultCode::UNKNOWN;
+
+  // Main loop runs at 10hz
+  timer_ = this->create_wall_timer(1000ms, std::bind(&WaypointsNavigation::execLoop, this));
+  waitActionServer();
+  RCLCPP_INFO(this->get_logger(), "Successfully connected to server");
+  RCLCPP_INFO(this->get_logger(), "Waiting for waypoint navigation to start");
+}
+
+bool WaypointsNavigation::startNavCallback(const std::shared_ptr<std_srvs::srv::Trigger::Request>& /*request*/,
+                                           std::shared_ptr<std_srvs::srv::Trigger::Response> responce)
```

Comment:
```suggestion
bool WaypointsNavigation::startNavCallback(const std::shared_ptr<std_srvs::srv::Trigger::Request>& /*request*/,
                                           std::shared_ptr<std_srvs::srv::Trigger::Response> response)
```
上記はtypoだと思うので修正お願いします。